packages/core/src/routes/logto-config.ts

```diff
@@ -34,4 +44,108 @@ export default function logtoConfigRoutes<T extends AuthedRouter>(
+      const configs = await getOidcConfigs();
+
+      // Remove actual values of the private key from response
+      const resultWithoutPrivateKey = Object.fromEntries(
+        Object.entries(configs).map(([key, value]) => [
+          key,
+          value.map(({ id, createdAt }) => ({ id, createdAt })),
+        ])
+      );
+
+      ctx.body = resultWithoutPrivateKey;
+
+      return next();
+    }
+  );
+
+  router.delete(
+    '/configs/:configKey/:keyId',
+    koaGuard({
+      params: z.object({
+        configKey: z.nativeEnum(LogtoOidcConfigKey),
+        keyId: z.string(),
+      }),
+      status: [204, 404, 422],
+    }),
+    async (ctx, next) => {
+      const { configKey, keyId } = ctx.guard.params;
+      const { rows } = await getRowsByKeys([configKey]);
+
+      const result = z.array(oidcPrivateKeyGuard).safeParse(rows[0]?.value);
+
+      if (!result.success) {
+        throw new RequestError({ code: 'request.general', status: 422 });
+      }
+
+      if (result.data.length <= 1) {
+        throw new RequestError({ code: 'oidc.key_required', status: 422 });
+      }
+
+      if (!result.data.some(({ id }) => id === keyId)) {
+        throw new RequestError({ code: 'oidc.key_not_found', id: keyId, status: 404 });
+      }
+
+      const updatedConfigValue = result.data.filter(({ id }) => id !== keyId);
+
+      await updateOidcConfigsByKey(configKey, updatedConfigValue);
+      await envSet.load();
+
+      ctx.status = 204;
+
+      return next();
+    }
+  );
+
+  router.post(
+    '/configs/rotate-oidc-keys',
+    koaGuard({
+      body: z.object({
+        privateKeyType: z.nativeEnum(PrivateKeyType).optional(),
+      }),
+      status: [204, 422],
+    }),
+    async (ctx, next) => {
+      const { privateKeyType } = ctx.guard.body;
+
+      const { rows } = await getRowsByKeys(Object.values(LogtoOidcConfigKey));
+
+      const existingPrivateKeys = rows.find(({ key }) => key === LogtoOidcConfigKey.PrivateKeys)
+        ?.value;
+      const existingCookieKeys = rows.find(({ key }) => key === LogtoOidcConfigKey.CookieKeys)
+        ?.value;
+
+      const privateKeyParseResult = z.array(oidcPrivateKeyGuard).safeParse(existingPrivateKeys);
+      const cookieKeyParseResult = z.array(oidcPrivateKeyGuard).safeParse(existingCookieKeys);
+
+      if (!privateKeyParseResult.success || !cookieKeyParseResult.success) {
+        throw new RequestError({ code: 'request.general', status: 422 });
+      }
+
+      const newCookieKey = generateOidcCookieKey();
+      const newPrivateKey = await generateOidcPrivateKey(privateKeyType);
+      const updatedPrivateKeys = [newPrivateKey, ...privateKeyParseResult.data].slice(0, 2);
+      const updatedCookieKeys = [newCookieKey, ...cookieKeyParseResult.data].slice(0, 2);
+
+      await Promise.all([
+        updateOidcConfigsByKey(LogtoOidcConfigKey.PrivateKeys, updatedPrivateKeys),
+        updateOidcConfigsByKey(LogtoOidcConfigKey.CookieKeys, updatedCookieKeys),
+      ]);
```

Comment:
is it possible not to rotate these two keys simultaneously?

Reply:
Yes, we are going to separate them, so that one can only rotate one type of key for each api call